File: backend/fastapi/app/domain/court.py
from datetime import datetime
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel
from app.domain.enums.general_status import GeneralStatus
from app.domain.enums.court_surface import CourtSurface


class Court(BaseModel):
    id: int
    slug: str
    name: str
    location_details: Optional[str] = None
    img_url: Optional[str] = None
    price_h: Decimal
    capacity: int
    is_indoor: bool
    surface: CourtSurface
    status: GeneralStatus
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: backend/policourt-backend/src/main/java/com/policourt/api/court/domain/model/Court.java
package com.policourt.api.court.domain.model;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

import com.policourt.api.court.domain.enums.CourtSurfaceEnum;
import com.policourt.api.shared.enums.GeneralStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Court {
    private UUID id;
    private String slug;
    private String name;
    private String locationDetails;
    private String imgUrl;
    private BigDecimal priceH;
    private Integer capacity;
    private Boolean isIndoor;
    private CourtSurfaceEnum surface;
    private GeneralStatus status;
    private Boolean isActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
